src/bson.rs

```diff
@@ -3,3 +3,82 @@
 // Because BSON depends on the protocol version,
 // need to develop an Enum<Struct, Struct, ...> to
 // work based on the protocol version.
+use bson::spec::BinarySubtype;
+use bson::{bson, Binary, Bson, Document};
+use serde::ser::{Serialize, SerializeStruct, Serializer};
+use serde::Deserialize;
+use std::io::{Error, ErrorKind};
+
+#[derive(Deserialize, PartialEq, Debug)]
+struct AccVersion1 {
+    v: i32,
+    //other stuff
```

Comment:
in the future - we might want to cache the AES key - provides an additional means of verification. just noting this for the future tho, not for this PR, that would be a good v2